File: backend/internal/models/order.go
package models

type Order struct {
	BaseModel
	DistributorID uint        `json:"distributorId"` // Foreign key
	CreatedBy     Distributor `gorm:"foreignKey:DistributorID" json:"createdBy"`
	CustomerID    uint        `json:"customerId"` // Foreign key
	RequestedFor  Customer    `gorm:"foreignKey:CustomerID" json:"requestedFor"`
	Status        string      `gorm:"not null;type:varchar(20)" json:"status"`
	TotalPrice    float64     `gorm:"not null" json:"totalPrice"`
	TotalWeight   float64     `gorm:"not null" json:"totalWeight"`
	TransportFare float64     `gorm:"not null" json:"transportFare"`
	WorkspaceID   uint        `json:"workspaceId"`
	Workspace     Workspace   `gorm:"foreignKey:WorkspaceID" json:"workspace"`
	OrderItems    []OrderItem `gorm:"foreignKey:OrderID" json:"orderItems"`
}

File: frontend/src/types/order.ts
import type { APIResponse, FilterResponse } from '.';
import type { Distributor } from './distributor';
import type { Tile } from './tile';
import type { Workspace } from './workspace';

export enum OrderStatus {
  Pending = 'pending',
  Approved = 'approved',
  Delivered = 'delivered',
  Cancelled = 'cancelled',
  All = '',
}

export enum DeliveryStatus {
  Pending = 'pending',
  InTransit = 'in-transit',
  Delivered = 'delivered',
  Cancelled = 'cancelled',
}

export type OrderResponse = FilterResponse<{
  price_lists: Order[];
}>;

export type FilterOrderResponse = APIResponse<OrderResponse>;

export type BulkDeleteOrderResponse = APIResponse<{
  ids: string[];
}>;

export type OrderItem = {
  id: number;
  createdAt: string;
  updatedAt: string;
  deletedAt?: string;
  orderId: number;
  unitPrice: number;
  totalPrice: number;
  tileId: number;
  tile: Tile;
  quantity: number;
  weight: number;
  totalWeight: number;
  distributorId: number;
  createdBy: Distributor;
  workspaceId: number;
  workspace: Workspace;
};

export type Order = {
  distributorId: number;
  createdBy: Distributor;
  customerId: number;
  status: OrderStatus;
  totalPrice: number;
  totalWeight: number;
  transportFare: number;
  workspaceId: number;
  workspace: Workspace;
  orderItems: OrderItem[];
  deliveryAddress: string;
  deliveredAt?: string; // ISO timestamp string
  deliveryStatus: DeliveryStatus;
  notes: string;
  discount: number;
  title: string;
  id: number;
  createdAt: string;
  updatedAt: string;
  deletedAt?: string;
};

export interface OrderFilter {
  search: string;
  status: OrderStatus;
}

export type OrderSortField = 'name' | 'status' | string;
